aten/src/ATen/BatchedFallback.cpp

```diff
@@ -123,12 +123,14 @@ void batchedTensorForLoopFallback(const c10::OperatorHandle& op, torch::jit::Sta
       // We assume that torch::jit::Stack is backed by vector<IValue> for
       // simplicity. When that is not the case, this code should be updated.
       const auto& argument = (*stack)[arguments_begin + arg_idx];
-      if (arg_idx != *batched_tensor_inputs_pos_iter) {
+      if (batched_tensor_inputs_pos_iter == batched_tensor_inputs_position.end()
+          || arg_idx != *batched_tensor_inputs_pos_iter) {
         // argument isn't a BatchedTensor
         torch::jit::push(stack, argument);
         continue;
       }
       // argument is a BatchedTensor
+      TORCH_INTERNAL_ASSERT(input_physical_views_iter != input_physical_views.end());
```

Comment:
This doesn't require a similar branch to above to fix UBN?

Reply:
Actually the condition above is sufficient to prevent UB in the bottom code. This is because:
- The batched_tensor_inputs_pos vector is the same size as the `input_physical_views` vector
- We always advance both iterators by 1 together (see lines 136, 137)
- If `batched_tensor_inputs_pos_iter == batched_tensor_inputs_position.end()`, then we know there are no more elements left in `input_physical_view_iter` anymore.
- If `batched_tensor_inputs_pos_iter == batched_tensor_inputs_position.end()` is True, then we'll never hit this code.
- Conversely, If `batched_tensor_inputs_pos_iter == batched_tensor_inputs_position.end()` is False, then we know there are still elements left in `input_physical_views_iter`, and so it is safe to dereference that pointer